contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1427 @@
+        address[] memory _operators
+    ) external override {
+        notify(_amount, _rewardMultiplier, _notifier, _operators);
+    }
+
+    /// @notice Takes the given number of queued slashing operations and
+    ///         processes them. Receives 5% of the slashed amount.
+    ///         Executes `involuntaryAuthorizationDecrease` function on each
+    ///         affected application.
+    function processSlashing(uint256 count) external override {
+        require(
+            slashingQueueIndex < slashingQueue.length && count > 0,
+            "Nothing to process"
+        );
+
+        uint256 maxIndex = slashingQueueIndex + count;
+        maxIndex = Math.min(maxIndex, slashingQueue.length);
+        count = maxIndex - slashingQueueIndex;
+        uint96 tAmountToBurn = 0;
+
+        uint256 index = slashingQueueIndex;
+        for (; index < maxIndex; index++) {
+            SlashingEvent storage slashing = slashingQueue[index];
+            tAmountToBurn += processSlashing(slashing);
+        }
+        slashingQueueIndex = index;
+
+        uint256 tProcessorReward = uint256(tAmountToBurn).percent(
+            SLASHING_REWARD_PERCENT
+        );
+        notifiersTreasury += tAmountToBurn - tProcessorReward.toUint96();
+        emit SlashingProcessed(msg.sender, count, tProcessorReward);
+        if (tProcessorReward > 0) {
+            token.safeTransfer(msg.sender, tProcessorReward);
+        }
+    }
+
+    //
+    //
+    // Auxiliary functions
+    //
+    //
+
+    /// @notice Returns the authorized stake amount of the operator for the
+    ///         application.
+    function authorizedStake(address operator, address application)
+        external
+        view
+        override
+        returns (uint96)
+    {
+        return operators[operator].authorizations[application].authorized;
+    }
+
+    /// @notice Returns staked amount of T, Keep and Nu for the specified
+    ///         operator.
+    /// @dev    All values are in T denomination
+    function stakes(address _operator)
+        external
+        view
+        returns (
+            uint96 tStake,
+            uint96 keepInTStake,
+            uint96 nuInTStake
+        )
+    {
+        OperatorInfo storage operator = operators[_operator];
+        tStake = operator.tStake;
+        keepInTStake = operator.keepInTStake;
+        nuInTStake = operator.nuInTStake;
+    }
+
+    /// @notice Returns start staking timestamp for T stake.
+    /// @dev    This value is set at most once, and only when a stake is created
+    ///         with T tokens. If a stake is created from a legacy stake,
+    ///         this value will remain as zero
+    function getStartTStakingTimestamp(address operator)
+        external
+        view
+        returns (uint256)
+    {
+        return operators[operator].startTStakingTimestamp;
+    }
+
+    /// @notice Returns staked amount of NU for the specified operator
+    function stakedNu(address operator)
```

Comment:
We need this function for Nu to check staked amount in legacy contract. Without this method we would need 2 points of Threshold network instead of 1: one call to get stake and another to convert using `VendingMachine`. So if it's not necessery for Keep side I'd have only one this method

Reply:
Cool, thanks for the explanation!